File: algorithms/1_easy/C/01925_count_square_sum_triples/solution.cpp
class Solution {
public:
    int countTriples(int n) {
        int count{0};

        for (int a{1}; a <= n; ++a) {
            for (int b{a}; b <= n; ++b) {
                const int c_square{a * a + b * b};
                const int possible_c = sqrt(c_square); 

                if (possible_c > n) {
                    break;
                }

                if (possible_c * possible_c == c_square) {
                    count += 2;
                }
            }
        }

        return count;
    }
};

File: algorithms/1_easy/C/01925_count_square_sum_triples/solution.rs
impl Solution {
    pub fn count_triples(n: i32) -> i32 {
        let mut count = 0_i32;

        for a in 1..=n {
            for b in a..=n {
                let c_square = a * a + b * b;
                let possible_c = (c_square as f64).sqrt() as i32; 

                if possible_c > n {
                    break;
                }

                if possible_c * possible_c == c_square {
                    count += 2;
                }
            }
        }

        count
    }
}
